doc/usecases.md

```diff
@@ -0,0 +1,188 @@
+#Layer2
+
+Usage can refer to this [doc](layer2.md)
+
+## Precondition
```

Comment:
Precondition  => Prerequiste, `Precondition` seems not that common-used in technical doc.